src/common/snippets/include/snippets/lowered/runtime_configurator.hpp

```diff
@@ -0,0 +1,198 @@
+namespace lowered {
+
+/**
+ * @interface RuntimeConfigurator
+ * @brief Describes the runtime-dependent (shape-dependent) information using Runtime Config
+ */
+class RuntimeConfigurator {
+    class LoopInitializer;
+    class FirstLoopInitializer;
+    class MainLoopInitializer;
+    class LastLoopInitializer;
+
+public:
+    RuntimeConfigurator();
+
+    /**
+     * @brief Initialize config using LinearIR state
+     * @param linear_ir the updated LinearIR
+     * @return config with runtime parameters
+     */
+    const RuntimeConfig& init(const lowered::LinearIR& linear_ir);
+    /**
+     * @brief Update config using LinearIR state
+     *        Note: Must be called only on static shapes!
+     * @param linear_ir the updated LinearIR
+     * @return config with runtime parameters
+     */
+    const RuntimeConfig& update(const lowered::LinearIR& linear_ir);
+    /**
+     * @brief Clone with new LinearIR info with loop descs, data offsets savings
+     * @param linear_ir the updated LinearIR
+     */
+    std::shared_ptr<RuntimeConfigurator> clone(const lowered::LinearIR& linear_ir);
+    /**
+     * @brief Reset config: remove all loop descriptors
+     */
+    void reset();
+
+private:
+    using LinearIR = lowered::LinearIR;
+    /**
+     * @brief Initialize the map of loops descriptors using LoopManager of LinearIR at first time
+     * @param loop_manager LoopManager of needed LinearIR
+     */
+    void init_loop_descriptors(const LinearIR::LoopManagerPtr& loop_manager);
+    /**
+     * @brief Update the existing the map of loops descriptors using LoopManager of LinearIR
+     * @param loop_manager LoopManager of needed LinearIR
+     */
+    void update_loop_descriptors(const LinearIR::LoopManagerPtr& loop_manager);
+    /**
+     * @brief Initialize input and output data offsets
+     */
+    void init_data_offsets();
+    /**
+     * @brief Calculate offset for input or output of body
+     * @param desc port descriptor
+     * @param data_size byte size of data type of input or output
+     * @param is_input true if it's input otherwise it's false
+     * @param rank common tensor rank
+     * @param offsets reference on the target offsets for update
+     */
+    inline void offset_calculation(const lowered::PortDescriptorPtr& desc, size_t data_size, bool is_input, size_t rank, std::vector<size_t>& offsets);
+    /**
+     * @brief Initialize input and output of LinearIR descriptors and data sizes.
+     * @param linear_ir target linear ir
+     */
+    void init_io_info(const LinearIR& linear_ir);
+
+    size_t m_io_num = 0;
+    size_t m_in_num = 0;
+    size_t m_tensor_rank = 0;
+    std::vector<PortDescriptorPtr> m_io_descs = {};
+    std::vector<size_t> m_io_data_sizes = {};
+
+    std::map<RuntimeConfig::LoopDescriptor::Type, std::shared_ptr<RuntimeConfigurator::LoopInitializer>> m_desc_initializers;
+    RuntimeConfig m_config;
+    bool m_inited = false;
+};
+
+
+class RuntimeConfigurator::LoopInitializer {
```

Comment:
We have a special class responsible for loop management `LinearIR::LoopManager`. It owns information about all loops, it has their loop_info, which in turn has all the information about the loop properties and handlers used to process specific iterations. 
The question is do we really need to introduce 4 additional classes just to copy a couple fields from loop_info to LoopDescriptor and occasionally call some `init_` functions (again, some updates based on loop_info)?
Can we instead create a ` RuntimeConfig::LoopDescriptor::update(const LoopInfoPtr& loop_info)` that will update the LoopDescriptor based on loop_info?
Another suggestion is to create `LoopDescriptor` from loop_info directly, and store the pointer to "bounded" loop_info there. In this case you will be able to call `update()` without any arguments, and it will do all the same work without extra fuss.

Reply:
Removed `LoopDescriptor` and `LoopInitializer` classes at all.
Now we work with `ExpandedLoopInfo`